web/src/APIClient.ts

```diff
@@ -1,11 +1,15 @@
-import { ListDashboardResponse } from './gen/pb/api/v1/data.js';
+import {
+  ListDashboardResponse,
+  ListDashboardGroupResponse,
+} from './gen/pb/api/v1/data.js';
 
 export interface APIClient {
   getDashboards(): Array<String>;
+  getDashboardGroups(): Array<String>;
 }
 
 export class APIClientImpl implements APIClient {
```

Comment:
I added this initially to do testing across the API. With the way you've done testing (modifying the properties of the view and seeing how it reacts), I don't think this is a useful abstraction anymore. In a later PR, we should consider deleting this object.

Reply:
> I added this initially to do testing across the API. With the way you've done testing (modifying the properties of the view and seeing how it reacts), I don't think this is a useful abstraction anymore. In a later PR, we should consider deleting this object.

> We are not using the added functionality to the API Client. Better to remove it altogether

I've added independent functions in `TestgridIndex.ts`, `APIClient.ts` can be removed.